### DISCUSSION TITLE
Select2

### DISCUSSION THREAD

**Discussion by Martin-1182** [livewire/livewire#4776] [1. Help]
>How to return saved values for editing from select2? I have no problem saving, but when trying to edit, the selector does not show the saved value. If I do it using with basic select, I get the value correctly, but it doesn’t work with select2. Any help would be helpful. Thank you!
``` 
<div wire:ignore>
      <select
          id="suppliers" 
          class="select2">
             @foreach ($allSuppliers as $supplier)
                <option value="{{ $supplier['id'] }}">{{ $supplier['name'] }}</option>
             @endforeach
      </select>
 </div>

@push('scripts')
    <script>
         document.addEventListener('livewire:load', () => {
            let suppliers = $('#suppliers')
            initSelect()

            Livewire.hook('message.processed', (message, component) => {
                initSelect()
            })

            Livewire.on('setSuppliersSelect', values => {
                suppliers.val(values).trigger('change.select2')
            })

            suppliers.on('change', function(e) {
                @this.set('voucher.supplier_id', suppliers.select2('val'))
            })

            function initSelect() {
                suppliers.select2({
                    placeholder: '{{ __('Select your option') }}',
                    allowClear: !suppliers.attr('required'),
                })
            }

        })
    </script>
@endpush
```

**Comment by joshhanley** [comment]
@Martin-1182 using Select2 with Livewire can be a pain to get working.

I'd recommend searching through Livewire's issues and discussions for other peoples Select2 implementations so you can have a look at how they are doing it, and maybe just use their implementation.

I personally don't use Select2 though, so can't be of much help.

**Reply by thanjeys** [reply]
Hi, Is there any alternative for select2, I used with Alpine also, got same issue.

**Reply by ph7jack** [reply]
@thanjeys @Martin-1182  I've create a package of components for the TallStack. You can use the select component from wireui
https://livewire-wireui.com/docs/select

**Reply by thanjeys** [reply]
Hi @PH7-Jack , wire-ui is fantastic package. But our application already developed in Bootstrap only.  Can i use Select search option only with Bootstrap ? 

Thanks
